common/io.hpp

```diff
@@ -90,8 +90,8 @@ class Discard : virtual public Writer {
 
 class StringReader : virtual public Reader {
 private:
-	StreamReader reader_;
 	std::stringstream s_;
+	StreamReader reader_;
```

Comment:
How is this a fix for a warning?

Reply:
haha, I know it looks weird. But it will be a lesson to you, same as it was to me :)

Ordering mattes appareantly. So now the declaration matches the initialization order in the constructor, because this is the order in which they are initialized.

Reply:
The reason it matters is that destruction is the reverse of the member order, and it's good practice that construction and destruction are always 100% inverse of each other.

Reply:
So kudos to the compiler here :1st_place_medal:

Reply:
Also, that makes a lot of sense actually!

Reply:
Ehm, I don't agree that it makes much sense because it's supposed to be a declaration and thus, by definition, declarative, i.e. with order not being important. :wink: But thank for the clarification.